embulk-output-redshift/src/main/java/org/embulk/output/RedshiftOutputPlugin.java

```diff
@@ -83,6 +83,10 @@ public interface RedshiftPluginTask extends AwsCredentialsTaskWithPrefix, Plugin
         @ConfigDefault("true")
         public boolean getDeleteS3TempFile();
 
+        @Config("max_s3_upload_threads_per_task")
+        @ConfigDefault("null")
```

Comment:
The default value of the original PR is `5`. Why you change this value?
https://github.com/embulk/embulk-output-jdbc/pull/311/files#diff-3ad638974347d49a3df0afc40dc2fa00382b91105aa42eeec25468daccbfc434R54

Any update after this comment in trocco servcie?
https://github.com/embulk/embulk-output-jdbc/pull/311

> As newCachedThreadPool would create new thread for execution if and only if current thread are all occupied, it seems tenable to pass a maximum of thread to newCachedThreadPool. However, due to the execution mechanism of embulk is ansync we may have troubles in how to determine the max of it. The most simple way to resolve is just make the maximum of each thread pool as a argument. After some performance test, it seems fine to make the default value of thread_maximum as 5.

Reply:
I've kept the original behavior by default, due to the requested in the following comment.

https://github.com/embulk/embulk-output-jdbc/pull/311#pullrequestreview-1241537103
```
It changes the thread behavior even if thread_maximum is not specified.
newCachedThreadPool() and newFixedThreadPool(5) behave different.
Can you make the it Optional<Interger>, and keep the behavior when the option is not specified?
```

> Any update after this comment in trocco servcie?

No, the maximum number of threads remains `5` in the trocco.

Reply:
OK, Thanks. Do you think whether is is better to change `5` by default in future release?

Reply:
I don't think there is a problem with the change, but I don't think it needs to be changed.
I think users would change the configuration as needed.